geoapps/contours/application.py

```diff
@@ -172,88 +158,25 @@ def update_name(self, _):
-            **{
-                "axis": axes(),
-                "resolution": self.resolution.value,
-                "window": {
-                    "center": [self.window_center_x.value, self.window_center_y.value],
-                    "size": [self.window_width.value, self.window_height.value],
-                    "azimuth": self.window_azimuth.value,
-                },
-                "contours": get_contours(
-                    self.interval_min.value,
-                    self.interval_max.value,
-                    self.interval_spacing.value,
-                    self.fixed_contours.value,
-                ),
-            },
-        )
+            param_dict["geoh5"] = workspace
 
-        if contour_set is not None:
-            vertices, cells, values = [], [], []
-            count = 0
-            for segs, level in zip(contour_set.allsegs, contour_set.levels):
-                for poly in segs:
-                    n_v = len(poly)
-                    vertices.append(poly)
-                    cells.append(
-                        np.c_[
-                            np.arange(count, count + n_v - 1),
-                            np.arange(count + 1, count + n_v),
-                        ]
-                    )
-                    values.append(np.ones(n_v) * level)
-                    count += n_v
-            if vertices:
-                vertices = np.vstack(vertices)
-                if self.z_value.value:
-                    vertices = np.c_[vertices, np.hstack(values)]
-                else:
-                    if isinstance(entity, (Points, Curve, Surface)):
-                        z_interp = LinearNDInterpolator(
-                            entity.vertices[:, :2], entity.vertices[:, 2]
-                        )
-                        vertices = np.c_[vertices, z_interp(vertices)]
-                    else:
-                        vertices = np.c_[
-                            vertices,
-                            np.ones(vertices.shape[0]) * entity.origin["z"],
-                        ]
+            if self.live_link.value:
+                param_dict["monitoring_directory"] = self.monitoring_directory
 
-            temp_geoh5 = f"{entity.name}_{data[0].name}_{time():.3f}.geoh5"
-            with self.get_output_workspace(
-                self.export_directory.selected_path, temp_geoh5
-            ) as workspace:
-                curve = Curve.create(
-                    workspace,
-                    name=string_name(self.export_as.value),
-                    vertices=vertices,
-                    cells=np.vstack(cells).astype("uint32"),
-                )
-                out_entity = curve
-                if len(self.ga_group_name.value) > 0:
-                    out_entity = ContainerGroup.create(
-                        workspace, name=string_name(self.ga_group_name.value)
-                    )
-                    curve.parent = out_entity
+            self.params.update(param_dict)
```

Comment:
Unfortunetly just updating params causes issues for multiple clicks. The workspace ends up closing and returning an error. Let's use the same mechanics as in the iso_surface and create a fresh Params from dict. Something like:

```
ifile = InputFile(
  ui_json=self.params.input_file.ui_json,
  validation_options={"disabled": True},
)

new_params = ContoursParams(input_file=ifile, **param_dict)
new_params.write_input_file()
driver = ContoursDriver(new_params)
```